store/src/main/java/org/apache/rocketmq/store/timer/TimerMessageStore.java

```diff
@@ -1140,8 +1146,8 @@ public MessageExtBrokerInner convertMessage(MessageExt msgExt, boolean needRoll)
         } else {
             msgInner.setTopic(msgInner.getProperty(MessageConst.PROPERTY_REAL_TOPIC));
             msgInner.setQueueId(Integer.parseInt(msgInner.getProperty(MessageConst.PROPERTY_REAL_QUEUE_ID)));
-            MessageAccessor.clearProperty(msgInner, MessageConst.PROPERTY_REAL_TOPIC);
-            MessageAccessor.clearProperty(msgInner, MessageConst.PROPERTY_REAL_QUEUE_ID);
+            // MessageAccessor.clearProperty(msgInner, MessageConst.PROPERTY_REAL_TOPIC);
```

Comment:
could you explain why comment these lines and how the second bug occurs?

Reply:
Alternatively, we should add a messageAccessior.deepCopyProperties instead of using the same reference.

Reply:
> Alternatively, we should add a messageAccessior.deepCopyProperties instead of using the same reference.

Your idea looks better, although I don't think commenting out this code will have any impact，so I modified the code as you suggested.